qemu/tests/virtio_mem_numa_basic.py

```diff
@@ -0,0 +1,69 @@
+import time
+
+from virttest import error_context
+
+from virttest.utils_misc import normalize_data_size
+from virttest.utils_misc import NumaInfo
+from provider import virtio_mem_utils
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Memory share and discard-data hotplug test
+    1) Boot guest with two virtio-mem devices
+    2) Check virtio-mem devices
+    3) Resize virtio-mem devices
+    4) Check virtio-mem devices
+    5) Resize virtio-mem devices to the maximum
+    6) Check virtio-mem devices
+    7) Resize virtio-mem devices to the minimum
+    8) Check virtio-mem devices
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    host_numa_node = NumaInfo()
+    node_list = host_numa_node.online_nodes_withmem
+    if len(node_list) < 2:
+        test.cancel("This host only has one NUMA node, skipping test...")
+
+    timeout = int(params.get("login_timeout", 240))
+    threshold = params.get_numeric("threshold", target_type=float)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=timeout)
+
+    virtio_mem_utils.check_memory_devices("requested-size_memory_vmem", params, threshold, vm, test)
+    virtio_mem_utils.check_numa_plugged_mem("requested-size_memory_vmem", params, threshold, session, vm, test)
+
+    requested_size_vmem0 = int(float(normalize_data_size(params.get("requested-size_memory-vmem0"), 'B')))
```

Comment:
This seems to be repeated many times, I don't know if we can write a loop.
like:
cfg:
`     requested-size_vmem0 = "1G 4G 8G 0"`
py:
```
     for requested-size in requested-size_vmem0.split():
          vm.monitor.qom_set()
          time.sleep(30)
```

Reply:
Changed to a loop, now the auto case logic has changed a little but I think it shouldn't be relevant. Previously every virtio-mem device was resized and then checked its value, now, a virtio-mem device is resized + checked all the needed times, then the same with the other one.